modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheProcessor.java

```diff
@@ -921,7 +921,7 @@ public void blockGateways() {
             GridCacheAffinityManager aff = cache.context().affinity();
 
             if (aff != null)
-                aff.cancelFutures();
+                aff.onKernalStop(true);
```

Comment:
@agoncharuk .... Or we can just remove aff.cancelFutures(); Because further we call onKernalStop(true); in a loop which eventually delegate cancelFutures(); ?